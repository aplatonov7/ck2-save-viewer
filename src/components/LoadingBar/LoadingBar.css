.LoadingBar__container {
  width: 100%;
  height: .5rem;
  overflow: hidden;
  background-color: #c5cae9;
}

.LoadingBar__indicator {
  width: 100%;
  height: 100%;
  background-color: #3f51b5;
  transition: transform 0.2s ease;
}