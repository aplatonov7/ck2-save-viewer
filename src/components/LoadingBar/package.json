{
  "name": "LoadingBar",
  "version": "0.0.0",
  "private": true,
  "main": "./LoadingBar.js"
}