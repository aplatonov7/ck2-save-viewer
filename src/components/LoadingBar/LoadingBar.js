import React from 'react'
import PropTypes from 'prop-types'

import './LoadingBar.css'

const LoadingBar = ({ value }) => (
  <div className="LoadingBar__container">
    <div
      className="LoadingBar__indicator"
      style={{ transform: `translateX(${-100 + (value * 100)}%)` }}
    />
  </div>
)

LoadingBar.propTypes = {
  value: PropTypes.number,
}

LoadingBar.defaultProps = {
  value: 0,
}

export default LoadingBar
