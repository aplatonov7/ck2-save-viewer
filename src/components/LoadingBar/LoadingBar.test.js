import React from 'react'
import LoadingBar from './LoadingBar'

describe('LoadingBar component', () => {
  let wrapper

  const props = {
    value: 0,
  }

  beforeEach(() => {
    wrapper = shallow(<LoadingBar {...props} />)
  })

  it('should set indicator styles according to the value prop', () => {
    expect(wrapper.find('.LoadingBar__indicator').props().style).toEqual({
      transform: `translateX(${-100 + 0 * 100}%)`,
    })
    wrapper.setProps({ value: 0.5 })
    expect(wrapper.find('.LoadingBar__indicator').props().style).toEqual({
      transform: `translateX(${-100 + 0.5 * 100}%)`,
    })
    wrapper.setProps({ value: 1 })
    expect(wrapper.find('.LoadingBar__indicator').props().style).toEqual({
      transform: `translateX(${-100 + 1 * 100}%)`,
    })
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<LoadingBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
