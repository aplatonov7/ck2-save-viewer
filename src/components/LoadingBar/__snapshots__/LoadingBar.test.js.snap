// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingBar component should match snapshot 1`] = `
<div
  className="LoadingBar__container"
>
  <div
    className="LoadingBar__indicator"
    style={
      Object {
        "transform": "translateX(-100%)",
      }
    }
  />
</div>
`;
